src/Zicht/Bundle/UrlBundle/Aliasing/Listener.php

```diff
@@ -147,76 +168,102 @@ protected function isExcluded($url)
      */
     public function onKernelRequest(Event\GetResponseEvent $event)
     {
-        if ($event->getRequestType() === HttpKernelInterface::MASTER_REQUEST) {
-            $request = $event->getRequest();
-            $publicUrl = rawurldecode($request->getRequestUri());
+        if ($event->getRequestType() !== HttpKernelInterface::MASTER_REQUEST) {
+            return;
+        }
 
-            if ($this->isExcluded($publicUrl)) {
-                // don't process urls which are marked as excluded.
-                return;
-            }
+        $request = $event->getRequest();
+        $publicUrl = rawurldecode($request->getRequestUri());
 
-            if ($this->isParamsEnabled) {
-                if (false !== ($queryMark = strpos($publicUrl, '?'))) {
-                    $originalUrl = $publicUrl;
-                    $publicUrl = substr($originalUrl, 0, $queryMark);
-                    $queryString = substr($originalUrl, $queryMark);
-                } else {
-                    $queryString = null;
-                }
+        if ($this->isExcluded($publicUrl)) {
+            // don't process urls which are marked as excluded.
+            return;
+        }
 
-                $parts = explode('/', $publicUrl);
-                $params = array();
-                while (false !== strpos(end($parts), '=')) {
-                    array_push($params, array_pop($parts));
-                }
-                if ($params) {
-                    $publicUrl = join('/', $parts);
+        $queryString = '';
+        if (false !== ($queryMark = strpos($publicUrl, '?'))) {
+            $queryString = substr($publicUrl, $queryMark);
+            $publicUrl = substr($publicUrl, 0, $queryMark);
+        }
 
-                    $parser = new UriParser();
-                    $request->query->add($parser->parseUri(join('/', array_reverse($params))));
+        if ($this->isParamsEnabled) {
+            $parts = explode('/', $publicUrl);
+            $params = array();
+            while (false !== strpos(end($parts), '=')) {
+                array_push($params, array_pop($parts));
+            }
+            if ($params) {
+                $publicUrl = join('/', $parts);
 
-                    if (!$this->aliasing->hasInternalAlias($publicUrl, false)) {
-                        $this->rewriteRequest($event, $publicUrl . $queryString);
+                $parser = new UriParser();
+                $request->query->add($parser->parseUri(join('/', array_reverse($params))));
```

Comment:
Aha... eerdere comment is met deze `array_reverse` teniet gedaan.  Echter... waarom dan niet de code hetschrijven zodat het dezelfde volgorde blijft hebben?  Lijkt mij minder foutgevoelig...

Reply:
Ik denk dat we dit wel even als een los issue moeten behandelen. Dit is al bestaande code, ik heb alleen de indenting aangepast van dit stuk (de Git diff snapt er alleen niks meer van :smile:)

Reply:
Nu ik er wat meer ben ingedoken zie ik dat dit geen issues op kan leveren. Op het moment van omgekeerd inlezen, wordt dit niet als key => value opgeslagen, maar als indexed array met string waarden "key=value". Later wordt inderdaad `array_reverse()` toegepast:

```
$publicUrl = '/path/foo=A/foo=B';
// resulteert in:
$params = ['foo=B', 'foo=A'];
join('/', array_reverse($params)) geeft 'foo=A/foo=B'
```